ordered_product/tests/test_ordered_product_page.py

```diff
@@ -0,0 +1,188 @@
+        assert template_names == set()
+
+    @pytest.mark.django_db()
+    def test_display_new_order_page(self, client, client0_login):
+        response = client.get('/ordered_product/new_order/')
+        assert response.status_code == 200
+        template_names = set(template.origin.template_name for template in response.templates)
+        assert 'ordered_product/new_order.html' in template_names
+
+    @pytest.mark.django_db()
+    def test_valid_search_product(self, client, saved_supplier_product0, client0_login):
+        VALID_PRODUCT_NAME = saved_supplier_product0.qr_code.product_name
+        product = Product.objects.get(product_name=VALID_PRODUCT_NAME)
+        supplier_products_by_objects_filter = SupplierProduct.objects.filter(qr_code=product)
+        assert saved_supplier_product0 in supplier_products_by_objects_filter
+        response = client.get('/ordered_product/new_order/search_product', {'product_name': VALID_PRODUCT_NAME})
+        assert response.status_code == 200
+        assert response.context['message'] == ""
+        supplier_products_by_the_web_view = response.context['supplier_products']
+        assert list(supplier_products_by_the_web_view) == list(supplier_products_by_objects_filter)
+
+    @pytest.mark.django_db()
+    def test_invalid_search_product(self, client, client0_login):
+        INVALID_PRODUCT_NAME = 'Eggplant'
+        with pytest.raises(ObjectDoesNotExist):
+            Product.objects.get(product_name=INVALID_PRODUCT_NAME)
+        response = client.get('/ordered_product/new_order/search_product', {'product_name': INVALID_PRODUCT_NAME})
+        assert response.status_code == 200
+        assert response.context['message'] == "No products found."
+        assert response.context['supplier_products'] == []
+
+    @pytest.mark.django_db()
+    def test_valid_find_delivery(self, client, saved_client0, saved_supplier0, supplier_product0, delivery_location0,
+                                 client0_login):
+        supplier_product0.save()
+        delivery_location0.save()
+        assert supplier_product0.user_name == saved_supplier0
+        assert delivery_location0.user_name == saved_supplier0
+        response = client.get('/ordered_product/new_order/find_delivery/'+str(supplier_product0.supplier_product_id))
+        assert response.status_code == 200
+        template_names = set(template.origin.template_name for template in response.templates)
+        assert 'ordered_product/new_order.html' in template_names
+        assert response.context['sup_pro_id'] == supplier_product0.supplier_product_id
+        delivery_location_by_objects_filter = DeliveryLocation.objects.filter(
+                                                          user_name=supplier_product0.user_name,
+                                                          location=saved_client0.area)
+        assert delivery_location0.location != saved_client0.area
+        assert list(delivery_location_by_objects_filter) == []
```

Comment:
I think the test needs to be simplified. I'm unable to understand what we're testing here and why we end up not finding anything.

Reply:
I change line 128 to:
`assert delivery_location0 not in delivery_location_by_objects_filter`